src/lib.rs

```diff
@@ -451,7 +451,49 @@ fn _encode(input: &[u8], options: Options) -> lib::String {
         } else {
             was_cr = false;
         }
-        encode_byte(&mut result, byte, &mut on_line, limit, &mut backup_pos);
+
+	// look for runs of characters that don't need encoding - this
+	// is very common (QP is normally used on ASCII text where
+	// most characters don't need encoding) and much faster than
+	// calling encode_byte on each character.  To keep this from
+	// completely reimplmenting append(), only do this if we have
+	// at least 3 characters left on the line and don't try to
+	// deal with the line-ending stuff.
+	if limit - on_line >= 3 && !needs_encoding(byte) {
```

Comment:
I feel like this should be `>` rather than `>=`. Can you write a test that enters this branch when `limit -on_line == 3` and the very next byte is one that needs encoding?

Reply:
Added a test to that effect.  I think the output is correct, but off-by-ones are always hard to feel confident about!  One thing I'd like to do is add some fuzz tests - those found some interesting edge cases in zoomascii and they might here too.